1-src/api/auth/authorization.mts

```diff
@@ -24,9 +24,11 @@ export const jwtAuthenticationMiddleware = async(request: JWTRequest, response:
 
     //Inject jwtUserId into request object
     else {
-        request.jwt = request.headers['jwt'];
-        request.jwtUserId = 0;
-        request.jwtUserRole = RoleEnum.STUDENT;
+        const client_token = request.headers['jwt'];
+        const token_data = getJWTData(client_token);
+        request.jwt = client_token
+        request.jwtUserId = token_data['userID'];
+        request.jwtUserRole = token_data['userRole'];
 
         next();
     }
```

Comment:
Please implement **Line #53**: We need to compare the JWT user_id against the provided userId,  Currently once, a JWT is created it can be used by any user to gain access.
_Here's a route to test: http://localhost:5000/api/user/profile/12 
headers: {
user-id: 1 
jwt: ...
}
//Note: Here profile 1 which owns the JWT, is viewing profile 12.  (That's allowed to view each others profiles)_

Reply:
Done